### ISSUE TITLE
Implement XEP-0084: User Avatar

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vacuum-im/vacuum-im#330]
```
Right now, Vacuum-IM implements 2 types of Avatar:
1. XEP-0008: IQ-Based Avatars
2. XEP-0153: vCard-Based Avatars

Both of them are historical, so there is no garantee that they won't be 
deprecated/obsolete later. The 1st one is Deferred now, so its implementation 
is not recommended.
Moreover, those XEPs are traffic-consuming, 'cause you either has to retreive 
your current vCard(Avatar) to calculate its SHA-1 hash value, or store avatar 
image data locally and send it to every client which needs it!
Also, you'll spam all the contacts in your roster with those hash values in 
your presence, every time you change your status, even if their clients do not 
support those XEPs!

The mordern (and the mostly recommened) way to exchange Avatars is PEP-based 
XEP-0084: User Avatar. This XEP do not have weak points, mentioned obove. I 
think, this it should be implemented along with other PEP-based features (see 
Issue: 20)

```

Original issue reported on code.google.com by `yagimo...@gmail.com` on 4 Jan 2011 at 4:49
- Blocked on: #510
